server/config/config.go

```diff
@@ -91,3 +91,11 @@ func (c *Config) Validate() error {
 
 	return nil
 }
+
+// Ensure that *Config implements ConfigProvider interface.
+var _ ConfigProvider = &Config{}
+
+// GetConfig implements ConfigProvider.
+func (c *Config) GetConfig() (*Config, error) {
+	return c, nil
```

Comment:
We could also call the `Validate` method here and bubble up any error, but I figure that makes more sense for the server to do at startup rather than push the responsibility into every ConfigProvider implementation.